app/forms/idv/in_person/residential_address_form.rb

```diff
@@ -0,0 +1,40 @@
+module Idv
+  module InPerson
+    class ResidentialAddressForm
+      include ActiveModel::Model
+      include Idv::InPerson::ResidentialAddressValidator
+
+      ATTRIBUTES = %i[residential_state residential_zipcode residential_city residential_address1
+                      residential_address2].freeze
+
+      attr_accessor(*ATTRIBUTES)
+
+      # TODO: complete below in another story - data persistence is out of scope for 9002
```

Comment:
I wasn't sure of the best way to communicate that data persistence is out of scope.  I can delete the TODO comment, if people are concerned about clogging up the code base with to-dos.

Reply:
I think that it will work better in the long run if we don't introduce this new ResidentialAddressForm and instead keep using the existing AddressForm to collect residential address. And then update the [StateIdForm](https://github.com/18F/identity-idp/blob/9128f75bf312670e3b3d6fb6a866009825e1e273/app/forms/idv/state_id_form.rb#L6-L7) so that it uses a different set of keys, like `state_id_address1`, `state_id_address2`, etc.

So when the user enters their residential address we just call it `address`. And when the user enters their state ID address we called it `state_id_address`. And then if they indicate that they don't have a separate residential address then perhaps we just copy all of the `state_id_address1` etc. fields into `address1`

That would keep our data model more consistent with the remote proofing flow. And if we introduce this new `residential_address1` etc. stuff then we need to update the USPS, PhoneFinder, GPO, and sharing-PII-back-to-the-partner logic to conditionally read from either `address1` or `residential_address1` depending on if the user has a different residential address or not

It's not called out in the AC for the ticket but I think if we merge this as-is it'll mean more rework in future tickets :(

Now that we're here I'm feeling like we should have done a mini tech-spec on these session fields to flesh out details like this before we started working on these tickets

Reply:
I think if we went with my suggestion then we also wouldn't want a separate ResidentialAddress page or step. Instead the address page would change its appearance/content depending on whether the feature flag was set

And if the feature flag is set, and the user indicates on the state ID page that their residential address does match their ID, then we would just skip the address page (eg mark it as complete)